ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -486,11 +505,13 @@ public void terminateQueryForEntity(final String entity) {
 
     if (query.isPresent()) {
       final PersistentQueryMetadata metadata = query.get();
-      log.info("Terminating persistent query {}", metadata.getId());
+      log.info("Terminating persistent query {}", metadata.getQueryId());
       metadata.close();
-      persistentQueries.remove(metadata.getId());
+      persistentQueries.remove(metadata.getQueryId());
       livePersistentQueries.remove(metadata);
       allLiveQueries.remove(metadata);
+      referentialIntegrityTable
+          .removeQueryFromReferentialIntegrityTable(metadata.getQueryId().getId());
```

Comment:
I think we are missing a call to this from removeTemporaryQuery. Without it temporary queries will count as references to sources even after they terminate

Reply:
*missing a call to removeQueryFromReferentialIntegrityTable

Reply:
Never mind. I see we aren't tracking references for transient queries.